### DISCUSSION TITLE
Observe and WindowResized

### DISCUSSION THREAD

**Discussion by Naarakah** [bevyengine/bevy#14589] [Q&A] [answered]
Hi,

I have a problem with getting `WindowResized` events with the observe method. The following code with an `EventReader` works fine (prints when the windows is resized)

```rust
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_systems(Update, on_window_resized)
        .run();
}

fn on_window_resized(events: EventReader<WindowResized>) {
    for _ in events.read() {
        info!("window resized!");
    }
}
```

but with `observe`, nothing happens
```rust
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .observe(on_window_resized)
        .run();
}

fn on_window_resized(_: Trigger<WindowResized>) {
    info!("window resized!");
}
```

I'm learning bevy so surely I have made a mistake, but I can't find what it is.

Thanks

**Comment by Jondolf** [comment, marked as answer]
Buffered events (using `EventReader`/`EventWriter`) are currently treated separately from observer events. So `my_event_writer.send(event)` adds buffered events that can only be read by an `EventReader`, not observers. `WindowResized` is a buffered event like this, and can not be listened to by observers.

I believe there are some plans to further unify buffered events and observers in some way, and to make the API slightly clearer / less footgunny, but I don't remember details of what it'll look like.

Also see `Observer` docs: <https://docs.rs/bevy_ecs/latest/bevy_ecs/observer/struct.Observer.html>

**Reply by Naarakah** [reply]
Okay, thank you! I missed that information in my search.
So there's currently no way to react to a window resize without polling every frame?

**Reply by Michae1Weiss** [reply]
> Okay, thank you! I missed that information in my search. So there's currently no way to react to a window resize without polling every frame?

I found this in the official examples:
```rust
/// This system shows how to respond to a window being resized.
fn on_resize_system(
    mut resize_reader: MessageReader<WindowResized>,
) {
    for e in resize_reader.read() {
        // When resolution is being changed
    }
}
```
**Version**: Bevy 0.18
## Refs
[**bevy.org** - **Window / Window Resizing** example](https://bevy.org/examples/window/window-resizing/)
